src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs

```diff
@@ -1601,14 +1586,13 @@ private static bool AllCharsInVectorAreAscii<T>(Vector128<T> vector)
             if (typeof(T) == typeof(byte))
             {
                 return
-                    Sse41.IsSupported ? Sse41.TestZ(vector.AsByte(), Vector128.Create((byte)0x80)) :
+                    Sse41.IsSupported ? (vector.AsByte() & Vector128.Create((byte)0x80)) == Vector128<byte>.Zero :
```

Comment:
Similar here, I'd just make this the default path and only continue specializing Arm64.

We want to eventually move everything to this core pattern and have the JIT recognize and do the right thing.

Reply:
These changes can be made in a follow-up PR.